From b2c1f0e95c5f1406073b2b527b2b92e32d843888 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 3 Jul 2020 11:12:24 +0800
Subject: [PATCH] policy/modules/services/sendmail: make sendmail_t domain MLS
 trusted for reading from files up to its clearance

Allow sendmail_t to search /run/systemd/journal

Fixes:
avc:  denied  { search } for  pid=327 comm="newaliases" name="journal"
dev="tmpfs" ino=10990
scontext=system_u:system_r:sendmail_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/sendmail.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/sendmail.te b/policy/modules/services/sendmail.te
index d81d2d637..25f4834ac 100644
--- a/policy/modules/services/sendmail.te
+++ b/policy/modules/services/sendmail.te
@@ -126,6 +126,8 @@ mta_manage_spool(sendmail_t)
 mta_read_config(sendmail_t)
 mta_sendmail_exec(sendmail_t)
 
+mls_file_read_to_clearance(sendmail_t)
+
 optional_policy(`
 	cfengine_dontaudit_write_log_files(sendmail_t)
 ')
-- 
2.17.1

